csrc/host_ir/executor.cpp

```diff
@@ -751,8 +751,11 @@ void HostIrEvaluator::handle(kir::Allocate* allocate) {
 }
 
 void HostIrEvaluator::handle(HirAliasSelect* hir_alias_select) {
-  auto index =
-      expr_evaluator_.evaluate(hir_alias_select->index()).as<int64_t>();
+  auto indexed_id =
+      hir_alias_select->in()->getLogicalDomain().at(hir_alias_select->axis());
```

Comment:
A logical domain may have reduction dimensions, which don't exist in the corresponding at::Tensor. So it's likely problematic to use the same `int axis` on both `in()->getLogicalDomain()` and `input`, unless `in()` is guaranteed to be reduction free.